Tutorial review (#3605)
canonical/mir: doc/sphinx/tutorial/learn-what-mir-can-do.md

# Getting started with Mir

This tutorial will guide you through Mir's basic functionality. 

By the end of the tutorial, you will install and run a demo application, learn how to use Mir in different environments, and learn of Mir features that can be used by Mir-based compositors. 

If you are unfamiliar with Mir, start with this tutorial, and then proceed to a [developer tutorial](write-your-first-wayland-compositor) which will guide you through the process of writing compositors.

## Installing demo applications
Mir demos are available on Debian derivatives, Fedora, and Alpine. For distros
that don't have pre-built binaries, examples can be built from source.

### On Debian and its derivatives

```sh
sudo apt install mir-demos mir-graphics-drivers-desktop
```

### On Fedora
```sh
sudo dnf install mir-demos
```

### On Alpine
```sh
sudo apk add mir-demos mir
```

## Running applications on X11 or Wayland

You will work with a `miral-app` script that runs a shell with basic GUI.

> Mir allows you to run applications in the *shell mode* or in *kiosk mode*.

Reviewer: It comes from the pre-existing text but "shell mode" is a bad term: all the modes are types of shell.

```suggestion
Mir allows you to run a graphical shell in *desktop mode* or in *kiosk mode*.
```

> Shell mode means that the application is opened in a floating window manager and you can move the window around the screen, maximize or minimize it.

Reviewer: ```suggestion
Desktop mode means that application windows are opened are "floating and you can move them around the screen, maximize or minimize them.
```

> Kiosk mode means that the application is opened in fullscrean mode. 

Reviewer: ```suggestion
Kiosk mode means that the application is opened in fullscreen mode. 
```

To run the script in the shell mode:
```sh
miral-app
```

To run the script in the kiosk mode:
```sh
miral-app -kiosk
```

> ## Running applications natively

Reviewer: ```suggestion
## Running shells natively
```

> Mir compositors supports running application natively by launching them from a virtual terminal or a login screen.

Reviewer: ```suggestion
Mir compositors supports running shells natively by launching them from a virtual terminal or a login screen.
```

### Launching from a virtual terminal

> 1. Switch to a virtual terminal by pressing CTRL+ALT+F\<Number\> and log into the virstual environemtn.

Reviewer: ```suggestion
1. Switch to a virtual terminal by pressing CTRL+ALT+F\<Number\> and log in.
```

2. Run the script:
```sh
miral-app
```

### Launching from a login screen

Open the window manager list, start **Mir Shell**, log in and `miral-shell` will be running fullscreen.

## Using on-screen keyboards
Mir-based compositors support on-screen keyboards. 

> **Note**: Due to security reasons, the Wayland extensions are disabled by default. In this tutorial, you will override this setting by

Reviewer: ```suggestion
**Note**: Due to security reasons, some Wayland extensions needed by on-screen keyboards are disabled by default. In this tutorial, you will override this by
```
